src/app/dts/template-editor/template-editor.component.ts

```diff
@@ -38,7 +33,22 @@ export class TemplateEditorComponent implements OnInit {
     allowedContent: true,
     fullPage: true,
     startupMode: 'source',
-    height: '700px'
+    height: '700px',
+    on: {
+      // Fired after setting the editing mode. Cannot bind to the event in the template.
+      mode(event): void {
```

Comment:
I'm wondering if you could use the read-only mode for this:https://ckeditor.com/docs/ckeditor4/latest/guide/dev_angular.html#readonly but I'm not sure if that fits the requirements you have. 

It looks like a "change" event is sent only in WYSIWYG mode, so that could be a better option here: https://ckeditor.com/docs/ckeditor4/latest/guide/dev_angular.html#change

Basically, you could disable read-only when the "change" event is sent, but I'm not sure if that fits your exact requirements.

Reply:
Thanks for the feedback.  It's a tough issue and I appreciate the comments.

I don't that that option would fit our use case.  While the user needs editing abilities in both code and wysisyg modes, the user should only be able to save in wysisyg mode because that's the only time we're aware of changes and when we have access to the template data.  

The user should be able to edit in code mode, but not save.  To save, they need to preview the template, verify it's correct, and then save.

A problem with using the change event is that we we can't disable saving in code mode because the change event isn't fired. However I was able to use the JSON "mode" event to trap when the user changes to code mode, and then use the DOM to disable the save button.

Reply:
Ok, that's our only option then.